pkg/rootlessport/rootlessport_linux.go

```diff
@@ -248,6 +269,56 @@ outer:
 	return nil
 }
 
+func serve(listener net.Listener, pm rkport.Manager) {
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			// we cannot log this error, stderr is already closed
+			continue
+		}
+		ctx := context.TODO()
+		err = handler(ctx, conn, pm)
+		if err != nil {
+			conn.Write([]byte(err.Error()))
+		} else {
+			conn.Write([]byte("OK"))
+		}
+		conn.Close()
+	}
+}
+
+func handler(ctx context.Context, conn io.Reader, pm rkport.Manager) error {
+	var childIP string
+	dec := json.NewDecoder(conn)
+	err := dec.Decode(&childIP)
+	if err != nil {
+		return errors.Wrap(err, "rootlessport failed to decode ports")
+	}
+	portStatus, err := pm.ListPorts(ctx)
+	if err != nil {
+		return errors.Wrap(err, "rootlessport failed to list ports")
+	}
+	for _, status := range portStatus {
+		err = pm.RemovePort(ctx, status.ID)
+		if err != nil {
+			return errors.Wrap(err, "rootlessport failed to remove port")
+		}
+	}
+	// pm.RemovePort() does not guarantee that the port can be used again since it
+	// closes ports in another goroutine. Therefore we add a sleep to make sure
+	// the other goroutine can close the ports before we add them again.
+	time.Sleep(50 * time.Microsecond)
```

Comment:
Can we use channels for better robustness?

Reply:
Yes but we would need to fix this in rootlesskit so that `RemovePort()` guarantees that the connection was closed.

Reply:
PTAL https://github.com/rootless-containers/rootlesskit/pull/262

Reply:
Thanks, released RootlessKit v0.14.3